app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -469,13 +470,15 @@ private void setResultAndFinish(boolean goToJobInfo) {
 
     public void handleConnectButtonPress() {
         selectedAppIndex = -1;
-        ConnectManager.unlockConnect(this, success -> {
-            if(success) {
-                ConnectManager.goToConnectJobsList();
-                setResult(RESULT_OK);
-                finish();
-            }
-        });
+        Intent intent=new Intent(this, ConnectIdActivity.class);
+        startActivity(intent);

```

Comment:
Also, let's move the launcher into a function in ConnectManager like launchConnectId(). That way the LoginActivity and CommCareSetupActivity don't have to copy the code, and we'll have a single entry point for the activity.